graphrag/index/flows/create_base_text_units.py

```diff
@@ -65,21 +63,18 @@ async def create_base_text_units(
     chunked = chunked.explode("chunks")
     chunked.rename(
         columns={
-            "chunks": chunk_column_name,
+            "chunks": "chunk",
         },
         inplace=True,
     )
-    chunked["chunk_id"] = chunked.apply(
-        lambda row: gen_md5_hash(row, [chunk_column_name]), axis=1
+    chunked["id"] = chunked.apply(lambda row: gen_md5_hash(row, ["chunk"]), axis=1)
+    chunked[["document_ids", "chunk", "n_tokens"]] = pd.DataFrame(
+        chunked["chunk"].tolist(), index=chunked.index
     )
-    chunked[["document_ids", chunk_column_name, n_tokens_column_name]] = pd.DataFrame(
-        chunked[chunk_column_name].tolist(), index=chunked.index
-    )
-    chunked["id"] = chunked["chunk_id"]
+    # rename for downstream consumption
+    chunked.rename(columns={"chunk": "text"}, inplace=True)
 
-    output = cast(
-        pd.DataFrame, chunked[chunked[chunk_column_name].notna()].reset_index(drop=True)
-    )
+    output = cast(pd.DataFrame, chunked[chunked["text"].notna()].reset_index(drop=True))
```

Comment:
We can remove this cast by doing ```chunked = chunked.loc[chunked["text"].notna()].reset_index(drop=True)```

Reply:
I feel like we're finding a bunch of this. Thanks to pandas warnings, I think we've been doing it "wrong" by not using `loc`.

Reply:
This didn't quite work so I'm going to skip for now and add a v1 todo to remove as many casts as possible